NPMK/openNEV.m

```diff
@@ -835,14 +839,23 @@
     end % end if ~isempty(allExtraDataPacketIndices)
 
     clear Timestamp tRawData count idx;
-      
-   % now read waveform
-    fseek(FID, Trackers.fExtendedHeader + 12, 'bof'); % Seek to location of spikes
-    fseek(FID, (Trackers.readPackets(1)-1) * Trackers.countPacketBytes, 'cof');
-    NEV.Data.Spikes.WaveformUnit = Flags.waveformUnits;
-    NEV.Data.Spikes.Waveform = fread(FID, [(Trackers.countPacketBytes-12)/2 Trackers.readPackets(2)], ...
-        [num2str((Trackers.countPacketBytes-12)/2) '*int16=>int16'], 12);
-    NEV.Data.Spikes.Waveform(:, [digserIndices allExtraDataPacketIndices]) = []; 
+
+    % now read waveform
+    if strcmpi(NEV.MetaTags.FileTypeID, 'NEURALEV')
```

Comment:
It looks like the only difference between the 2 blocks of code is the `8` and `12`. Ignoring the strcmpi issue for now, it would be better to do
`if strcmpi(NEV.MetaTags.FileTypeID, 'NEURALEV'); hoffset=8; elseif strcmpi(NEV.MetaTags.FileTypeID, 'BREVENTS'); hoffset=12; end;`
Then use hoffset in place of 8 or 12. This reduces duplication.

That being said, I think that doing a string comparison for every single spike is going to be rather slow. Is there any way to store `hoffset` in the `NEV` structure and just use that?

Reply:
This was a great suggestion and I modified the code to store the byte offset in a variable, then just have one block of code performing the file read.

On the second point, note that this code is not run once per spike packet, but once for all spike packets, so I don't think the performance of strcmpi will be a strong issue. That same comparison is used a couple times throughout the file and might benefit from an update to check once and store the result as you have suggested.